src/odemis/driver/pigcs.py

```diff
@@ -3572,19 +3574,19 @@ def sendQueryCommand(self, addr, com):
 
         if addr is None:
             full_com = "".join(com)
-            prefix = ""
+            prefix = b""
         else:
             full_com = "".join("%d %s" % (addr, c) for c in com)
-            prefix = "0 %d " % addr
+            prefix = b"0 %d " % addr
 
         with self.ser_access:
-            logging.debug("Sending: '%s'", full_com.encode('string_escape'))
-            self.serial.write(full_com)
+            logging.debug("Sending: '%s'", full_com)
```

Comment:
`%r`

Reply:
full_com should be a string by now, if the argument was a list, it's converted by `"".join()`.

Reply:
Let's escape it with `to_str_escape()`.

Reply:
Still needs an escape or a `%r`